ecommerce/src/main/kotlin/io/hhplus/ecommerce/checkout/application/usecase/CheckoutUseCase.kt

```
@@ -78,26 +80,57 @@ class CheckoutUseCase(
-                // 재고 가용성 확인
-                if (!inventoryDomainService.checkStockAvailability(item.productId, item.quantity)) {
-                    val availableQuantity = inventoryDomainService.getAvailableQuantity(item.productId)
+                // 재고 예약 (락 획득 + 가용성 검증 + 예약을 원자적으로 수행)
+                // 낙관적 락 충돌 시 최대 3회 재시도
+                var lastException: Exception? = null
+                repeat(3) { attempt ->
+                    try {
+                        inventoryDomainService.reserveStock(item.productId, item.quantity)
+                        lastException = null
+                        return@repeat
+                    } catch (e: io.hhplus.ecommerce.inventory.exception.InventoryException.InsufficientStock) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)